crates/extensions/c8y_firmware_manager/src/error.rs

```diff
@@ -42,6 +43,19 @@ pub enum FirmwareManagementError {
 
     #[error(transparent)]
     FromMqttError(#[from] tedge_mqtt_ext::MqttError),
+
+    #[error("Download from {firmware_url} failed with {err}")]
+    FromDownloadError {
+        firmware_url: String,
+        err: DownloadError,
+    },
+
+    #[error("Child device {child_id} did not respond within the timeout interval of {time_limit_sec}sec. Operation ID={operation_id}")]
+    FromTimeout {
```

Comment:
(trivial) I think we use the `From` prefix for the errors inheriting from somewhere. Hence, here you don't need `From`.

```suggestion
    ExceedTimelimit {
```

Reply:
Fixed: https://github.com/thin-edge/thin-edge.io/pull/2600/commits/e37bc2ed72ad973f4de621f34c588f0983c4a506